sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -1039,6 +1040,12 @@ def map_cel_vars(a):
             ],
             'taskSpec': template['spec'],
           }
+        if template['spec'].get("workspaces"):
+          workspaces_spec = []
+          for item in template['spec']['workspaces']:
+            workspaces_spec.append({"name": item["name"], "workspace": item["name"]})
```

Comment:
workspace's value?

Reply:
template['spec']['workspaces'] is a list of dict like below, on the pipelinetask level we just need to have the name of the workspace and which workspace we want to refer to in the pipeline definition. For simplicity, I expect all the user defined workspaces will have the same name within the same pipeline.

```
taskSpec:
  workspaces:
  - name: new-ws
    readOnly: true
```